### PR TITLE
Convert helpers/general and storage.js to ts

### PR DESCRIPTION
**Pull request by johnschoeman**
Why:
We would like to use typescript on some files which import theses
functions and constants.

This commit:
Updates the files to ts.  Some rather permissive types are being
returned from the functions, which could likely be tightened up in a
future commit.

### CODE REVIEW

**Comment by johnschoeman** [comment]
trying to ship smaller commits

File: app/helpers/General.ts [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
-    let data = await AsyncStorage.getItem(key);
+    const data = await AsyncStorage.getItem(key);
 
     if (isString) {
       return data;
     }
 
-    return JSON.parse(data);
+    if (data) {
+      return JSON.parse(data);
```
Comment (JacobJaffe) [inline app/helpers/General.ts:14]: Doesn't this return an any type? 
Without explicit setters / getters per payload to the store I think this whole `GetStoreData` needs to be typed as an `any` return

Comment (johnschoeman) [reply app/helpers/General.ts:14]: you are correct! good catch

File: app/TracingStrategyAssets.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import { useTranslation } from 'react-i18next';
```
Comment (JacobJaffe) [inline app/TracingStrategyAssets.ts]: It looks like to me this file got added without removing the corresponding .js version?

**Review by JacobJaffe — CHANGES_REQUESTED** [review]

**Review by JacobJaffe — APPROVED** [review]

**Review by tparesi — APPROVED** [review]
LGTM

**Pull request merged**